common/scala/src/main/scala/whisk/core/database/memory/MemoryArtifactStore.scala

```diff
@@ -0,0 +1,342 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.database.memory
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.ContentType
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.util.{ByteString, ByteStringBuilder}
+import spray.json.{DefaultJsonProtocol, DeserializationException, JsObject, JsString, RootJsonFormat}
+import whisk.common.{Logging, LoggingMarkers, TransactionId}
+import whisk.core.database.StoreUtils._
+import whisk.core.database._
+import whisk.core.entity._
+import whisk.http.Messages
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+import scala.util.Try
+
+object MemoryArtifactStoreProvider extends ArtifactStoreProvider {
```

Comment:
can you add a java/scala doc (comment) explaining the purpose of memory store provider; furthermore, do you anticipate this will be used outside of testing (in which case should this move to the tests package)?

Reply:
Would add docs. So far no concrete usecase to use them outside testing. One major reason to put them in main codebase was getting coverage data easily. 

Typically Code Coverage data is only calculated for production code but here I would like to track coverage of Memory store implementation logic such that it can be ensured that tests cover all aspects of the implementation